src/goto-instrument/accelerate/polynomial_accelerator.h

```diff
@@ -34,27 +34,21 @@ class polynomial_acceleratort:public path_accelerationt
 public:
   polynomial_acceleratort(
     message_handlert &message_handler,
-    const symbol_tablet &_symbol_table,
-    const goto_functionst &_goto_functions)
-    : message_handler(message_handler),
-      symbol_table(const_cast<symbol_tablet &>(_symbol_table)),
-      ns(symbol_table),
-      goto_functions(_goto_functions),
-      utils(symbol_table, message_handler, goto_functions, loop_counter)
-  {
-    loop_counter=nil_exprt();
-  }
-
-  polynomial_acceleratort(
-    message_handlert &message_handler,
+    const optionst &options,
     const symbol_tablet &_symbol_table,
     const goto_functionst &_goto_functions,
-    exprt &_loop_counter)
+    const exprt &_loop_counter = nil_exprt())
```

Comment:
What's this?

Reply:
Default argument removing the need for 2 separate functions.

Reply:
How is this related to this PR?

Reply:
I was adding an extra constructor argument. In this case it was clunky, as there was a default parameter implemented manually, rather than using the language construct. I rolled that into one to reduce PR's size.

Reply:
Ok, I would have preferred delegating constructors instead, out of some dislike for default arguments. But I guess I don't have solid facts to back this up. Just make sure the commit messages states includes a note about this refactoring.

Reply:
The change has been separated into another commit.